### PR TITLE
Enhancement/testing strategy backend

### PR DESCRIPTION
**Pull request by PinkusKSquare**
First version of test integration.

### CODE REVIEW

File: test/main.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
+// During the test the env variable is set to test
+process.env.NODE_ENV = "test";
```
Comment (mishelashala) [inline test/main.ts]: set this environment variable in the `.env` file, never modify global variables

File: test/main.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+  before(async function() {
+    // Open db connection if it is not currently open.
+    const isConnectionOpen = !!db.authenticate();
+    if (!isConnectionOpen) {
+      await db.sync();
+    }
+
+    // Before testing it is convenient to create a new user..
+    const createdUser: any = await User.create({ ...testUser, role: "admin" });
```
Comment (mishelashala) [inline test/main.ts]: you can avoid the `any` type if you explicitly add a return value type to `User.create`

File: test/main.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    const user: any = await User.findOne({
```
Comment (mishelashala) [inline test/main.ts]: you can avoid the `any` type if you explicitly add a return value type to `User.findOne`

File: test/v1/Auth.ts [hunk 4]
```diff
@@ -0,0 +1,399 @@
+import { chai, Credentials, User } from "../common";
+
+const apiPath = "http://localhost:8888/api/v1/auth/";
+
+export const authTest = (
+  auth: Credentials,
+  testUser: { email: string; password: string }
```
Comment (mishelashala) [inline test/v1/Auth.ts]: create a separate interface for this

File: test/common.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import { Room } from "../app/models/Room";
+import { User } from "../app/models/User";
+
+// Import the dev-dependencies
+import * as chai from "chai";
+import chaiHttp = require("chai-http");
+let should = chai.should();
+chai.use(chaiHttp);
+
+interface UserData {
```
Comment (mishelashala) [inline test/common.ts]: interfaces should start with capital `I` and be in `PascalCase`:
```ts
interface IUserData {}
```

File: test/common.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  profileId: string;
+  userId: string;
+}
+
+interface Credentials {
```
Comment (mishelashala) [inline test/common.ts]: interfaces should start with capital `I` and be in `PascalCase`:
```ts
interface ICredentials {}
```

File: test/v1/Auth.ts [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+            res.body.should.have.property("token");
+            res.body.should.have.property("expires");
+            res.body.should.have.property("refresh_token");
+            res.body.should.have.property("user");
+            res.body.should.have.property("profile");
+            done();
+          });
+      });
+
+      // it("it should not login if email is not registered", (done) => {
```
Comment (mishelashala) [inline test/v1/Auth.ts]: delete commented code or skip the test

```js
it.skip("should not login if the email is not registered", (done) => {
  // ...
})
```

File: test/v1/Auth.ts [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+            if (err) {
+              throw err;
+            }
+            res.should.have.status(401);
+            res.body.should.be.a("string").and.equal("Unauthorized");
+            done();
+          });
+      });
+
+      // it("it should not login if params are invalid", (done) => {
```
Comment (mishelashala) [inline test/v1/Auth.ts]: delete commented code or skip the test

```js
it.skip("...", () => {})
```

File: test/v1/Auth.ts [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+          .set("Authorization", auth.refreshToken)
+          .end((err, res) => {
+            if (err) {
+              throw err;
+            }
+            res.should.have.status(401);
+            res.body.should.be.an("object").and.deep.equal({
+              name: "Error",
+              message: "This Token is blacklisted"
+              // "message": "This token cannot be used for this request."
```
Comment (mishelashala) [inline test/v1/Auth.ts]: delete commented code

File: test/common.ts (hunk 5 above)
Comment (PinkusKSquare) [reply test/common.ts]: Replaced this interface with IGetUserParams which already exists in UserInterfaces.ts

File: test/v1/Auth.ts (hunk 4 above)
Comment (PinkusKSquare) [reply test/v1/Auth.ts]: New interface IUserLogin created at UserInterfaces.ts

File: test/v1/Auth.ts (hunk 7 above)
Comment (PinkusKSquare) [reply test/v1/Auth.ts]: Skipped test as it could/should be implemented future updates.

File: test/v1/Auth.ts (hunk 8 above)
Comment (PinkusKSquare) [reply test/v1/Auth.ts]: Skipped test as it could/should be implemented future updates.

File: test/common.ts (hunk 6 above)
Comment (PinkusKSquare) [reply test/common.ts]: Renamed it to ICredential

File: test/v1/Auth.ts (hunk 9 above)
Comment (PinkusKSquare) [reply test/v1/Auth.ts]: Deleted

**Pull request merged**